src/dcurl.c

```diff
@@ -60,26 +70,44 @@ static uv_sem_t notify_remote;
 
 bool dcurl_init()
 {
-    bool ret = true;
+    bool ret = false;
 
 #if defined(ENABLE_AVX)
-    ret &= registerImplContext(&PoWAVX_Context);
+    if (registerImplContext(&PoWAVX_Context)) {
+        initStatus = initStatus | AVX_BIT;
+        ret |= true;
```

Comment:
Valid value `1` specified in  `enum` would be much better for understanding purpose.